source/text/glyph.vert

```diff
@@ -5,25 +5,30 @@ precision lowp int;
 
 
 #if __VERSION__ == 100
-    #extension GL_EXT_draw_buffers : enable
-    attribute vec2 a_vertex;
-    /* [ texture ll: vec2, ur: vec2 ] */
-    attribute vec4 a_texCoord;
-    attribute vec3 a_origin;
-    attribute vec3 a_tangent;
-    attribute vec3 a_up;
+
+#extension GL_EXT_draw_buffers : enable
+attribute vec2 a_vertex;
+/* [ texture ll: vec2, ur: vec2 ] */
+attribute vec4 a_texCoord;
+attribute vec3 a_origin;
+attribute vec3 a_tangent;
+attribute vec3 a_bitangent;
+
 #else
-    layout(location = 0) in vec2 a_vertex;
-    /* [ texture ll: vec2, ur: vec2 ]*/
-    layout(location = 1) in vec4 a_texCoord;
-    layout(location = 2) in vec3 a_origin;
-    layout(location = 3) in vec3 a_tangent;
-    layout(location = 4) in vec3 a_up;
+
+in vec2 a_vertex;
+/* [ texture ll: vec2, ur: vec2 ]*/
+in vec4 a_texCoord;
+in vec3 a_origin;
+in vec3 a_tangent;
+in vec3 a_bitangent;
```

Comment:
I am not sure if `bitangent` is the correct word, see [bitanget](https://en.wikipedia.org/wiki/Bitangent). This is why, some time ago, we changed `bitan` to `up`.

Reply:
Reason was that those terms are related to the glyphs on the baseline, i.e., "tangent" is the vector tangential to the baseline, "up" is the vector orthogonal to the baseline. Since those terms are only used in typesetting space (which is 2D), we could change "up"/"bitangent" to "orthogonal", but this would merely be a mathematical description and would not contribute to the comprehension of the code.

"bitangent" would mean "tangential to the baseline in two (or more) points", which really does not fit our meaning here, as it is mathimatically incorrect.

Reply:
For consistency, in some docs we talk about "direction" and "up" (user perspective) and and some others we use "tangent" and "bitangent" (mathematical perspective).